crates/ruff/src/rules/refurb/rules/reimplemented_starmap.rs

```diff
@@ -0,0 +1,304 @@
+use ruff_diagnostics::{AutofixKind, Diagnostic, Edit, Fix, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{self as ast, Expr};
+use ruff_python_codegen::Generator;
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::{Ranged, TextRange};
+
+use crate::checkers::ast::Checker;
+use crate::importer::ImportRequest;
+use crate::registry::AsRule;
+
+/// ## What it does
+/// Checks for generator expressions, list and set comprehensions that can
+/// be replaced with `itertools.starmap`.
+///
+/// ## Why is this bad?
+/// When unpacking values from iterators to pass them directly to
+/// a function, prefer `itertools.starmap`.
+///
+/// Using `itertools.starmap` is more concise and readable.
+///
+/// ## Example
+/// ```python
+/// scores = [85, 100, 60]
+/// passing_scores = [60, 80, 70]
+///
+///
+/// def passed_test(score: int, passing_score: int) -> bool:
+///     return score >= passing_score
+///
+///
+/// passed_all_tests = all(
+///     passed_test(score, passing_score)
+///     for score, passing_score in zip(scores, passing_scores)
+/// )
+/// ```
+///
+/// Use instead:
+/// ```python
+/// from itertools import starmap
+///
+///
+/// scores = [85, 100, 60]
+/// passing_scores = [60, 80, 70]
+///
+///
+/// def passed_test(score: int, passing_score: int) -> bool:
+///     return score >= passing_score
+///
+///
+/// passed_all_tests = all(starmap(passed_test, zip(scores, passing_scores)))
+/// ```
+///
+/// ## References
+/// - [Python documentation: `itertools.starmap`](https://docs.python.org/3/library/itertools.html#itertools.starmap)
+#[violation]
+pub struct ReimplementedStarmap;
+
+impl Violation for ReimplementedStarmap {
+    const AUTOFIX: AutofixKind = AutofixKind::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Use `itertools.starmap` instead of the generator")
+    }
+
+    fn autofix_title(&self) -> Option<String> {
+        Some(format!("Replace with `itertools.starmap`"))
+    }
+}
+
+/// A abstract node that can be considered a candidate for replacement with `starmap`.
+pub(crate) trait StarmapEquivalent {
+    /// Get generated element.
+    fn get_element(&self) -> &Expr;
+    /// Get generator comprehensions.
+    fn get_generators(&self) -> &[ast::Comprehension];
```

Comment:
The get prefix isn't common in Rust. Instead, it is best practice to omit it
```suggestion
    fn element(&self) -> &Expr;
    /// Get generator comprehensions.
    fn generators(&self) -> &[ast::Comprehension];
```

Reply:
Will do